test/lib/next-test-utils.ts

```diff
@@ -171,7 +172,18 @@ export function renderViaHTTP(
 }
 
 export function findPort() {
-  return getPort()
+  // [NOTE] What are we doing here?
+  // There are some flaky tests failures caused by `No available ports found` from 'get-port'.
+  // This may be related / fixed by upstream https://github.com/sindresorhus/get-port/pull/56,
+  // however it happened after get-port switched to pure esm which is not easy to adapt by bump.
+  // get-port-please seems to offer the feature parity so we'll try to use it, and leave get-port as fallback
+  // for a while until we are certain to switch to get-port-please entirely.
+  try {
+    return getRandomPort()
+  } catch (e) {
+    require('console').warn('get-port-please failed, falling back to get-port')
+    return getPort()
```

Comment:
I don't see any reason why we need to fallback to `get-port`. If `get-port-please` does fail, IMHO we would assume there is no available port and the test should fail.

Furthermore, IMHO we should drop `get-port` in Next.js completely (our setup just don't work with pure ESM package, so why should we use pure ESM package when CJS/ESM dual package is available?) and switch its usage to `get-port-please`.

Reply:
Fallback is not meant to stay permanent, we will observe stability and eventually get rid of it.

Reply:
@kwonoj That makes sense. IMHO you should add a TODO comment here if it is not meant to be permanent.